### PR TITLE
DOC Improves input and target docstring for loss functions

### LINKED ISSUE #56581: [docs] Unclear docs of binary_cross_entropy / BCELoss
It says: `Function that measures the Binary Cross Entropy between the target and the output.`, but in fact it the argument names are `input` and `target`. It also does not say clearly if the inputs are probabilities, logits or log_probs. One can guess from the example, but much clearer to specify this explicitly. 

A lot of other methods like `kl_div` have the same problem

image

cc @brianjo @mruberry @albanD @jbschlosser

### PR DESCRIPTION
**Pull request by thomasjpfan**
Fixes https://github.com/pytorch/pytorch/issues/56581

### CODE REVIEW

File: torch/nn/modules/loss.py [hunk 1]
```diff
@@ -529,8 +529,8 @@ def forward(self, input: Tensor, target: Tensor) -> Tensor:
 
 
 class BCELoss(_WeightedLoss):
-    r"""Creates a criterion that measures the Binary Cross Entropy
-    between the target and the output:
+    r"""Function that measures the Binary Cross Entropy between the target and input
```
Comment (jbschlosser) [inline torch/nn/modules/loss.py:532]: "Function" isn't quite accurate here since it's a module- what do you think about just "Measures the..." ?

File: torch/nn/functional.py [hunk 2]
```diff
@@ -2908,8 +2909,8 @@ def binary_cross_entropy_with_logits(
     See :class:`~torch.nn.BCEWithLogitsLoss` for details.
 
     Args:
-        input: Tensor of arbitrary shape
-        target: Tensor of the same shape as input
+        input: Tensor of arbitrary shape as unnormalized scores
```
Comment (jbschlosser) [inline torch/nn/functional.py:2912]: maybe a comment like "unnormalized scores (often referred to as logits)" would be useful here and in `cross_entropy`

**Review by jbschlosser — COMMENTED** [review]
Good-looking updates! Couple suggestions but nothing big

File: torch/nn/modules/loss.py (hunk 1 above)
Comment (thomasjpfan) [reply torch/nn/modules/loss.py]: I mistakenly copied and pasted here, updated to PR to just replace "the output" with "the input probabilities".

**Review by jbschlosser — APPROVED** [review]
lgtm!
